pt_miniscreen/core/components/list.py

```diff
@@ -85,49 +114,51 @@ def _scroll_transition(self):
                 self.state.update({"transition_progress": progress + progress_step})
 
         active_transition = self.state["active_transition"]
-        row_to_remove = self.rows[0 if active_transition == "DOWN" else -1]
-        self.rows.remove(row_to_remove)
-        self.remove_child(row_to_remove)
+        if active_transition == "DOWN":
+            rows_to_remove = self.rows[:distance]
+        else:
+            rows_to_remove = self.rows[-distance:]
+
+        for row in rows_to_remove:
+            self.rows.remove(row)
+            self.remove_child(row)
```

Comment:
If visible_rows reflected the final rows array (see my other comment) then this could leverage it to be a little more concise and descriptive.

```suggestion
        for row in filter(lambda row: row not in self.visible_rows, self.rows):
            self.rows.remove(row)
            self.remove_child(row)
```